### DISCUSSION TITLE
Add button above Index views

### DISCUSSION THREAD

**Discussion by rctneil** [avo-hq/avo#1287] [General]
Hi,

I've just created an Export feature.  I would like to add an "Export" button above the Index view for that Resource.

Is this possible?

**Comment by adrianthedev** [comment]
Hey @rctneil.
We don't currently support customized controls on the index page as we do on [the show page](https://docs.avohq.io/2.0/customizable-controls.html#customize-the-controls), we have [that in the pipeline](https://github.com/avo-hq/avo/issues/1254), but you can create [a new action](https://docs.avohq.io/2.0/actions.html), and it will appear in the actions dropdown.

**Reply by rctneil** [reply]
Hi @adrianthedev 

Thanks for that.   I'm trying to create an Action and have the following:

```
class ExportCsv < Avo::BaseAction
  self.name = "Export CSV"
  self.may_download_file = true

  def handle(**args)
    models, fields, current_user, resource = args.values_at(:models, :fields, :current_user, :resource)

    redirect_to resources_people_export_path(format: :csv)

    succeed "Successfully exported People to CSV"
  end
end
```

Basically I have a route, that when called, provides a CSV file.   When I run the Action, I get the following error on the redirect_to line:

```
NoMethodError in Avo::ActionsController#handle
undefined method `resources_people_export_path' for #<ExportCsv:0x0000000123a2e698 @response={:messages=>[]}>
```

Any ideas how to get the Action to trigger the CSV download?

**Reply by adrianthedev** [reply]
I wouldn't go with this "redirect-to-route-that-downloads" path, but add the logic there (or run it through a service class) and then trigger the download from this action.
[Here](https://docs.avohq.io/2.0/actions.html#download) are the docs on how to trigger a download from an action. Also, I'd advise reading the whole document because it will give you a better understanding of what an action is and how it was meant to be used. I think you'll get good insights out of it.

Regarding the `NoMethodError`, you'd need to append `main_app` to that route helper. [This](https://docs.avohq.io/2.0/faq.html#why-don-t-regular-url-helpers-work-as-expected) is another bit from our documentation explaining what happens there and how engines work.
